modules/misc/listenbrainz.c

```diff
@@ -0,0 +1,585 @@
+    if ( !p_sys->p_current_song.psz_artist )
+    {
+        msg_Dbg (p_this, "Artist missing.");
+        DeleteSong (&p_sys->p_current_song);
+        b_skip = 1;
+    }
+
+    RETRIEVE_METADATA(p_sys->p_current_song.psz_title, Title);
+    if ( b_skip || !p_sys->p_current_song.psz_title )
+    {
+        msg_Dbg (p_this, "Track name missing.");
+        DeleteSong (&p_sys->p_current_song);
+        b_skip = 1;
+    }
+
+    if ( !b_skip )
+    {
+        RETRIEVE_METADATA(p_sys->p_current_song.psz_album, Album);
+        RETRIEVE_METADATA(p_sys->p_current_song.psz_musicbrainz_id, TrackID);
+        RETRIEVE_METADATA(p_sys->p_current_song.psz_track_number, TrackNum);
+        p_sys->p_current_song.i_length = SEC_FROM_VLC_TICK (input_item_GetDuration (item));
+        msg_Dbg (p_this, "Meta data registered");
+        vlc_cond_signal (&p_sys->wait);
+    }
+    vlc_mutex_unlock (&p_sys->lock);
+
+#undef RETRIEVE_METADATA
+}
+
+static listen_t CopySong (listen_t song)
+{
+    listen_t *copy = (listen_t *) malloc (sizeof (*copy));
+    memset (copy, 0, sizeof (*copy));
+    if ( song.psz_title )
+        copy->psz_title = strdup (song.psz_title);
+    if ( song.psz_artist )
+        copy->psz_artist = strdup (song.psz_artist);
+    if ( song.psz_album )
+        copy->psz_album = strdup (song.psz_album);
+    if ( song.psz_musicbrainz_id )
+        copy->psz_musicbrainz_id = strdup (song.psz_musicbrainz_id);
+    if ( song.psz_track_number )
+        copy->psz_track_number = strdup (song.psz_track_number);
+    copy->date = song.date;
+    return *copy;
+}
+
+static void Enqueue (intf_thread_t *p_this)
+{
+    bool b_skip = 0;
+    intf_sys_t *p_sys = p_this->p_sys;
+
+    vlc_mutex_lock (&p_sys->lock);
+
+    if ( !p_sys->p_current_song.psz_artist || !*p_sys->p_current_song.psz_artist ||
+         !p_sys->p_current_song.psz_title || !*p_sys->p_current_song.psz_title )
+    {
+        msg_Dbg (p_this, "Missing artist or title, not submitting");
+        b_skip = 1;
+    }
+
+    if ( p_sys->p_current_song.i_length == 0 )
+        p_sys->p_current_song.i_length = p_sys->time_played;
+
+    if ( !b_skip && p_sys->time_played < 30 )
+    {
+        msg_Dbg (p_this, "Song not listened long enough, not submitting");
+        b_skip = 1;
+    }
+
+    if ( !b_skip )
+    {
+        msg_Dbg (p_this, "Song will be submitted.");
+        listen_t p_copy_to_queue = CopySong (p_sys->p_current_song);
+        vlc_vector_push (&p_sys->queue, p_copy_to_queue);
+    }
+
+    vlc_cond_signal (&p_sys->wait);
+    DeleteSong (&p_sys->p_current_song);
+    p_sys->b_meta_read = false;
+    vlc_mutex_unlock (&p_sys->lock);
+}
+
+static void PlayerStateChanged (vlc_player_t *player, enum vlc_player_state state, void *data)
+{
+    intf_thread_t *intf = data;
+    intf_sys_t *p_sys = intf->p_sys;
+
+    if ( vlc_player_GetVideoTrackCount (player) )
+    {
+        msg_Dbg (intf, "Not an audio-only input, not submitting");
```

Comment:
You should remove this debug, you don't want this log to be displayed each time an user watch a video.